src/ensembl/src/content/app/custom-download/containers/content/result-loader/ResultLoader.tsx

```diff
@@ -57,64 +65,7 @@ const ResultHolder = (props: Props) => {
     props.setIsLoadingResult(false);
   }, [props.preview]);
 
-  if (props.preview && !props.preview.results && props.isLoadingResult) {
-    return (
-      <div className={styles.wrapper}>
-        <div className={styles.loaderWrapper}>
-          <CircleLoader />
-        </div>
-      </div>
-    );
-  }
-
-  if (!props.preview.results) {
-    return (
-      <div className={styles.wrapper}>
-        <div className={styles.noResult}>Sorry, no results</div>
-      </div>
-    );
-  }
-
-  const formattedResults = formatResults(
-    props.preview,
-    props.selectedAttributes
-  );
-  const headerRow = formattedResults.shift() || [];
-  return (
-    <>
-      {props.isLoadingResult && (
-        <div className={styles.loaderWrapper}>
-          <CircleLoader />
-        </div>
-      )}
-      <div className={styles.wrapper}>
-        {formattedResults.map((dataRow: string[], resultKey: number) => {
-          return (
-            <div key={resultKey} className={styles.resultCard}>
-              {headerRow.map((header: string, rowKey: number) => {
-                return (
-                  <div key={rowKey} className={styles.resultLine}>
-                    <div className={styles.lineHeader} title={header}>
-                      {header}
-                    </div>
-                    <div className={styles.lineValue}>
-                      {dataRow[rowKey] ? dataRow[rowKey] : '-'}
-                    </div>
-                  </div>
-                );
-              })}
-            </div>
-          );
-        })}
-      </div>
-    </>
-  );
-};
-
-type DispatchProps = {
-  fetchPreviewResult: (fetchPreviewResult: string) => void;
-  clearPreviewResult: () => void;
-  setIsLoadingResult: (setIsLoadingResult: boolean) => void;
+  return null;
```

Comment:
Huh? Is this component no longer needed? Should it be deleted?

Reply:
This is still responsible to load the results when the attributes or filters are changed. Not very sure if it needs to be inside an action on its own.

Reply:
Interesting. But using React’s component just for its `useEffect` hook without rendering anything is probably an abuse of React’s component model. I vaguely remember seeing react-router guys do something like that quite a while ago; but this probably can be expressed more idiomatically, without having a dedicated component which doesn’t render anything.

Reply:
Moved it to the Header component and removed ResultLoader.